### DISCUSSION TITLE
Add file with variables at build time?

### DISCUSSION THREAD

**Discussion by ssuess** [quasarframework/quasar#8978] [CLI - PWA mode] [answered]
I have a `version.txt` file that I include with my PWA and upload to the server to force a few things to happen during updates (and to make PWAs work they way they need to across all platforms). I have found that I can get it to be included in the build if I add it to the `public` folder, but I would like to insert one or more variables into it in a similar way that htmlVariables works in the index.template.html file. Obviously just including any of those in the file does not work. Do I need to use a separate post build, self-defined process like `afterBuild` described here: https://quasar.dev/quasar-cli/quasar-conf-js#property-build or is there an easier way to just replace some variables in this file? Thanks!

**Comment by ssuess** [comment]
Or is there some completely other process I should be using to add a file (that is my own) to the build output but needs some variables supplied at build time?

**Comment by ssuess** [comment, marked as answer]
Ok I figured this out. First install this package:

`npm i create-file-webpack --save-dev`

Then inside `quasar.conf.js` add this line at the top:

`const CreateFileWebpack = require('create-file-webpack')`

Then after `extendWebpack (cfg) {` add something like

```
extendWebpack (cfg) {
        cfg.plugins.push(new CreateFileWebpack({
          path: 'builds/prod',
          // file name
          fileName: 'version.txt',
          // content of the file
          content: '{ "version": "' + Date.now() + '" }'
        })),

```
Works great!

**Reply by jitendra1607** [reply]
if you just want to add existing file then add this script to package.json file
```
"postbuild": "run-s copyfile",
"copyfile": "cp -r /Applications/from_directory/version.txt /Applications/dist/version.txt"
```
